### PR TITLE
feat(Backend) : Implements vonneumann_entropy from tensorly to ivy/lineralg

### LINKED ISSUE #22049: vonneumann_entropy

### PR DESCRIPTION
**Pull request by Enigmafiend**
Close #22049

### CODE REVIEW

**Comment by Enigmafiend** [comment]
ivy-gardener

**Comment by Enigmafiend** [comment]
ivy-gardener

File: ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py [hunk 1]
```diff
@@ -1610,3 +1610,42 @@ def test_tucker_tensorly(tol_norm_2, tol_max_abs, shape, ranks):
         ivy.max(ivy.abs(rec_svd - rec_random)) < tol_max_abs,
         "abs norm of difference between svd and random init too high",
     )
+
+
+# von_entropy
+@st.composite
+def _get_dtype_and_square_matrix(draw):
+    dim_size = draw(helpers.ints(min_value=2, max_value=5))
+    dtype = draw(helpers.get_dtypes("float", full=True))
```
Comment (YushaArif99) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py:1619]: I think there are incompatible `dtypes` being generated here which are causing a few of the tests to fail with an error: 
```python
E   tensorflow.python.framework.errors_impl.InvalidArgumentError: Value for attr 'T' of int32 is not in the list of allowed values: bfloat16, half, float, double, complex64, complex128
E   	; NodeDef: {{node Sqrt}}; Op<name=Sqrt; signature=x:T -> y:T; attr=T:type,allowed=[DT_BFLOAT16, DT_HALF, DT_FLOAT, DT_DOUBLE, DT_COMPLEX64, DT_COMPLEX128]> [Op:Sqrt] name:
```
error log: [here](https://github.com/unifyai/ivy/actions/runs/6026565144/job/16349806736?pr=22521#step:3:448)

Maybe try using `helpers.get_dtypes("valid")` and see if that fixes the issue

**Review by YushaArif99 — CHANGES_REQUESTED** [review]
Left a comment 🙂

File: ivy/functional/ivy/experimental/linear_algebra.py [hunk 2]
```diff
@@ -201,9 +201,8 @@ def diagflat(
     ret
         The 2-D output array.
 
-    Functional Examples
```
Comment (YushaArif99) [inline ivy/functional/ivy/experimental/linear_algebra.py:204]: Could you kindly revert these changes 🙂

File: ivy/functional/ivy/experimental/linear_algebra.py [hunk 3]
```diff
@@ -1666,3 +1666,140 @@ def dot(
...
     return current_backend(a, b).dot(a, b, out=out)
+
+
+@handle_exceptions
+@handle_nestable
+@handle_array_like_without_promotion
+@inputs_to_ivy_arrays
+@handle_array_function
+@handle_device_shifting
+def general_inner_product(
```
Comment (YushaArif99) [inline ivy/functional/ivy/experimental/linear_algebra.py:1677]: where is this function being used?

**Review by YushaArif99 — CHANGES_REQUESTED** [review]
Hi @Enigmafiend 

Could you kindly address the abovementioned points? Also, could you kindly resolve the merge conflicts so that the CI tests can run? Thanks! 🙂

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Pull request closed without merging**